lesson34-arrays/index.js

```diff
@@ -0,0 +1,162 @@
+  { productName: "bread", qty: 2, bought: true, pricePerItem: 27, amount: 54 },
+];
+
+// 1. disply unbught and then unbought products
+// 1.1 - using filter, map and join
+// list of product showing as a String
+function sortProductByBought(array) {
+  const boughtProducts = array
+    .filter((product) => product.bought === true)
+    .map((product) => product.productName)
+    .join(", ");
+
+  const unboughtProducts = array
+    .filter((product) => product.bought === false)
+    .map((product) => product.productName)
+    .join(", ");
+
+  const sortedProductsString = `Unbought Products: ${unboughtProducts}. Bought Products: ${boughtProducts}`;
+  console.log({ sortedProductsString });
+  console.log(typeof sortedProductsString);
+}
+
+sortProductByBought(shopingList);
+
+// 1.2 - using filter, map and forEach
+function sortProductByBought1(array) {
+  const unboughtProducts = array
+    .filter((product) => product.bought === false)
+    .map((product) => product.productName);
+
+  console.log("Unbought Products:");
+  unboughtProducts.forEach((item) => {
+    console.log(item);
+  });
+
+  const boughtProducts = array
+    .filter((product) => product.bought === true)
+    .map((product) => product.productName);
+
+  console.log("Bought Products:");
+  boughtProducts.forEach((item) => {
+    console.log(item);
+  });
+}
+
+sortProductByBought1(shopingList);
+
+// 1.3 - using sort and map
+const sortProductByBought2 = (arrayOfProducts) => {
+  const sortedProducts = arrayOfProducts
+    .sort((a, b) => a.bought - b.bought)
+    .map((product) => product.productName);
+
+  return sortedProducts;
+};
+
+console.log(sortProductByBought2(shopingList));
+
+// 2.1 - The function takes the name of the product and marks it as purchased.
+function markProductPurchased(arrayOfProducts, nameOfProduct) {
+  return arrayOfProducts.map((product) => {
+    return product.productName === nameOfProduct ? { ...product, bought: true } : { ...product };
+  });
+}
+console.log(markProductPurchased(shopingList, "cheese"));
+console.log(markProductPurchased(shopingList, "orange"));
+
+//* NORMA
+/*
+1. Видалення продукту зі списку (видалення повинно проводитися шляхом створення нового масиву, 
+  в якому продукт, що ми шукаємо, буде відсутнім)*/
+
+const deleteProduct = (shopingList, nameOfproduct) => {
+  const newListOfProducts = [];
+  shopingList.forEach((product) => {
+    if (product.productName !== nameOfproduct) {
+      newListOfProducts.push(product);
+    }
+  });
+
+  return newListOfProducts;
+};
+
+console.log(deleteProduct(shopingList, "cheese"));
+
+/*2. Додавання покупки в список. Враховуй, що при додаванні покупки з уже існуючим в списку продуктом, 
+необхідно збільшувати кількість в існуючій покупці, а не додавати нову. При цьому також повинна 
+змінитися сума, наприклад, якщо ціна за одиницю 12, а кількості товарів стало 2, то сума буде 24
+*/
+function addProductToList(shopingList, nameOfproduct, qtyOfProduct, pricePerItem) {
```

Comment:
price per itm should to be count and receive rather only item name not quantity at etc 
as for instance console.log(addProductToList(shopingList, "milk", 3, 37));
{ productName: "milk", qty: 2, bought: true, pricePerItem: 35, amount: 70 },
milk per item here is 37 and above it is 35 so which one should be used ?